File: fsharp/clock/Clock.fs
module Clock

type Clock = {
    Minutes: int
}

let rec normalize (clock: Clock): Clock =
    // yesterday
    if clock.Minutes < 0 then
        { Minutes = clock.Minutes + (24 * 60) } |> normalize
    // tomorrow    
    else if clock.Minutes >= (24 * 60) then
        { Minutes = clock.Minutes % (24 * 60) } |> normalize
    // today    
    else
        clock

let mkClock (hours: int) (minutes: int): Clock =
    { Minutes = hours * 60 + minutes } |> normalize

let add (minutes: int) (clock: Clock): Clock =
    { Minutes = clock.Minutes + minutes } |> normalize

let subtract (minutes: int) (clock: Clock): Clock =
    { Minutes = clock.Minutes - minutes } |> normalize

let display (clock: Clock): string =
    let hours = clock.Minutes / 60
    let minutes = clock.Minutes % 60
    sprintf "%02i:%02i" hours minutes

File: csharp/clock/Clock.cs
using System;

public struct Clock
{
    private const int Day = 24 * 60;

    private int minutes;

    public Clock(int hours, int minutes)
    {
        this.minutes = minutes + (hours * 60);

        while (this.minutes < 0)
            this.minutes += Day;

        this.minutes %= Day;
    }

    public int Hours
    {
        get => this.minutes / 60;
    }

    public int Minutes
    {
        get => this.minutes % 60;
    }

    public Clock Add(int minutesToAdd)
    {
        return new Clock(0, this.minutes + minutesToAdd);
    }

    public Clock Subtract(int minutesToSubtract)
    {        
        return new Clock(0, this.minutes - minutesToSubtract);
    }

    public override string ToString()
    {
        return $"{this.Hours:00}:{this.Minutes:00}";
    }
}